### ISSUE TITLE
Add an organization filtering component

### ISSUE THREAD

**Issue by zugdev** [ubiquity/work.ubq.fi#128]
@0x4007 I know this was discussed previously and the "text field" can handle searching for a repo, but as an avid user of `work.ubq.fi` the lack of easy org filters actually bothers me a lot. I have implemented a pretty solution in around 4 hours, please give it a look.

[orgsSelector.webm](https://github.com/user-attachments/assets/57013b8f-174b-433d-b12e-34685b5ad37b)

If you like it lmk so I can put in more effort to polish and wrap it up. Open to feedbacks.

**Comment by zugdev** [comment]
Mobile view:

[mobileOrgSelector.webm](https://github.com/user-attachments/assets/7f54a239-94d1-4507-bae1-677c6bbdb454)

**Comment by 0x4007** [comment]
Hey, that looks nice! 

Im glad to hear you identify as an avid user and took it into your own hands to make a needed change. This is exactly what I hoped for in taking the open source strategy.  

My concern is that this approach doesn't scale. How will this be usable with 100+ partners? 

Maybe it makes sense to offer some more "first party support" for modded UIs (like we have official UIs directory for all of our projects) then we can see which ones are best based on popularity etc.  

### Suggestion

Ideally if there was a way to get a record of every task you were paid from, then we can save the amount of assigns per org in localStorage and sort by that order. 

We may be able to derive this from our permits saved in our database. Perhaps we can go by amount earned (which we will use as the basis of our XP score) which serves as a rough proxy for how involved you are within an org. 

That way, if you prefer to work on a specific org's tasks, they will appear in this carousel from most to least popular. 

The first time this page is visited or if the user is not logged in, we can hide this element. 

### XP

We have had this planned for almost a year but nobody started on it. Maybe this could be a catalyst to give it a shot to get started with.

**Comment by 0x4007** [comment]
@UbiquityOS what can you tell me about the planned XP features? Can you summarize all of the most recent plans?

**Comment by 0x4007** [comment]
Any update on getting this set up @gentlementlegen @sshivaditya2019

**Comment by gentlementlegen** [comment]
We noticed that Cloudflare workers are limited to 10 ms which is way too small to run the plugin so eventually this should be converted to an Action. @sshivaditya2019 you confirm?

**Comment by 0x4007** [comment]
Yes it should be an action. This is precisely the reason why we have first class support for actions. 

Shouldnt adapting it into an action be less restrictive than a worker, and as such, should be very simple?

**Comment by shiv810** [comment]
> We noticed that Cloudflare workers are limited to 10 ms which is way too small to run the plugin so eventually this should be converted to an Action. @sshivaditya2019 you confirm?

The workers were reaching the 10ms hard limit on CPU wall time. I’ve converted them into actions, but not using NCC, as that seems to cause some errors. 

@gentlementlegen, am I missing something here ? [Action Run](https://github.com/sshivaditya2019/command-ask/actions/runs/11418864599/job/31772801265)

**Comment by zugdev** [comment]
Yeah there is lots we can do to scale that. 

1. We can add a: "Recent" task on top of it and work on tracking the latest issues someone has delivered.

2. Another scaling option is to display a fixed amount of orgs, and for the last org to be a dropdown with all orgs and a search bar.

3. Sell relevance in the page? The highest the bid the first the project comes on the selector list.

Should I open a PR for the current component and later open another to handle scaling?

**Comment by zugdev** [comment]
Yeah there is lots we can do to scale that. 

1. We can add a: "Recent" task on top of it and work on tracking the latest issues someone has delivered.

2. Another scaling option is to display a fixed amount of orgs, and for the last org to be a dropdown with all orgs and a search bar.

3. Sell relevance in the page? The highest the bid the first the project comes on the selector list.

Should I open a PR for the current component and later open another to handle scaling?

**Comment by zugdev** [comment]
Another solution: I think every partner should be able to have a landing page on this website. Perhaps we change the header's logo when that happen, but they should be able to have a list of orgs they want to track. Therefore at `work.ubq.fi/aave` the only orgs showing would be aave related.

**Comment by 0x4007** [comment]
> Therefore at `work.ubq.fi/aave` the only orgs showing would be aave related.

This is a cool idea. I would like for now that it just loads the list of issues from that specific org. That way we can do a very simple dynamic setup. 

devpool.directory/aave would only load aave org tasks for example. 

I wonder if subdomain is cleaner but more complicated to setup. 

aave.devpool.directory

ubiquity-os-marketplace.devpool.directory

devpool.directory/ubiquity-os-marketplace

I suppose slash syntax looks better for long names.

**Comment by zugdev** [comment]
Can we push this? Do you want me to do the org component plus handling URLs? I am trying to filter the `ubiquity `org but as `ubiquity `is a substring `ubiquity-os` I just get all issues.

![image](https://github.com/user-attachments/assets/06ca2a78-01c6-4722-905f-b74915817521)

**Comment by 0x4007** [comment]
I suppose exact match only makes sense then

**Comment by zugdev** [comment]
So here we want a exact match option inside text field, correct?

**Comment by 0x4007** [comment]
No the url. /aave etc

**Comment by zugdev** [comment]
@gentlementlegen this one too! should we add a retry? or did the event not get to the kernel?

**Comment by gentlementlegen** [comment]
@zugdev not sure if we can add a retry because it's a worker so it cannot last very long at least on the free plan and would probably time out. Not sure why sometimes the fetch fails.

**Comment by zugdev** [comment]
> @zugdev not sure if we can add a retry because it's a worker so it cannot last very long at least on the free plan and would probably time out. Not sure why sometimes the fetch fails.

perhaps a fallback worker? triggered by fail? I am not familiar with how those plugins work yet, might be totally wrong

**Comment by gentlementlegen** [comment]
Since we are using workers on a free plan they are limited to something around [10 ms computation time](https://developers.cloudflare.com/workers/platform/limits/#worker-limits) and workers cannot spawn other workers if they are in the same account, so that's something we should investigate I suppose.
